core/templates/domain/statistics/LearnerActionObjectFactory.ts

```diff
@@ -67,24 +90,51 @@ export class LearnerAction {
   }
 }
 
+export class ExplorationStartLearnerAction extends
+  LearnerActionBase<'ExplorationStart'> {}
+
+export class AnswerSubmitLearnerAction extends
+  LearnerActionBase<'AnswerSubmit'> {}
+
+export class ExplorationQuitLearnerAction extends
+  LearnerActionBase<'ExplorationQuit'> {}
+
+export type LearnerAction = (
+  ExplorationStartLearnerAction |
+  AnswerSubmitLearnerAction |
+  ExplorationQuitLearnerAction);
+
 @Injectable({
   providedIn: 'root'
 })
 export class LearnerActionObjectFactory {
-  /**
-   * @property {string} actionType - type of an action
-   * @property {Object.<string, *>} actionCustomizationArgs - customization
-   *   dict for an action
-   * @property {number} [schemaVersion=LEARNER_ACTION_SCHEMA_LATEST_VERSION]
-   *   - schema version of the class instance.
-   * @returns {LearnerAction}
-   */
-  createNew(
-      actionType: string, actionCustomizationArgs: any,
-      schemaVersion: number): LearnerAction {
+  createNewExplorationStartAction(
+      actionType: 'ExplorationStart',
+      actionCustomizationArgs: IExplorationStartCustomizationArgs,
+      schemaVersion?: number): ExplorationStartLearnerAction {
     schemaVersion = schemaVersion ||
       StatisticsDomainConstants.LEARNER_ACTION_SCHEMA_LATEST_VERSION;
-    return new LearnerAction(
+    return new ExplorationStartLearnerAction(
+      actionType, actionCustomizationArgs, schemaVersion);
+  }
```

Comment:
There's no need for this argument, right? Let's just get rid of it and pass it directly from the implementation:

```suggestion
  createNewExplorationStartAction(
      actionCustomizationArgs: IExplorationStartCustomizationArgs,
      schemaVersion?: number): ExplorationStartLearnerAction {
    schemaVersion = schemaVersion ||
      StatisticsDomainConstants.LEARNER_ACTION_SCHEMA_LATEST_VERSION;
    return new ExplorationStartLearnerAction(
      'ExplorationStart', actionCustomizationArgs, schemaVersion);
  }
```

Likewise elsewhere.

Reply:
done